src/app/pages/registration/registration-page.component.ts

```diff
@@ -36,30 +33,22 @@ export class RegistrationPageComponent implements OnInit {
 
   registrationConfig: RegistrationConfigType;
 
-  fields$: Observable<FormlyFieldConfig[]>;
+  fields: FormlyFieldConfig[];
   model: Record<string, unknown>;
   options: FormlyFormOptions;
   form = new FormGroup({});
 
   ngOnInit() {
-    this.error$ = merge(this.accountFacade.userError$, this.accountFacade.ssoRegistrationError$);
-    this.fields$ = this.route.queryParamMap.pipe(
-      tap((paramMap: ParamMap) => (this.model = this.processParamMap(paramMap))),
-      map((paramMap: ParamMap) => ({
-        sso: paramMap.get('sso') === 'true',
-        userId: paramMap.get('userid'),
-        businessCustomer: this.featureToggle.enabled('businessCustomerRegistration'),
-      })),
-      tap(config => (this.registrationConfig = config)),
-      tap(
-        config =>
-          (this.options = this.registrationFormConfiguration.getRegistrationFormConfigurationOptions(
-            config,
-            this.model
-          ))
-      ),
-      map(config => this.registrationFormConfiguration.getRegistrationFormConfiguration(config))
+    this.error$ = this.registrationFormConfiguration.getErrorSources();
+
+    const snapshot = this.route.snapshot;
+    this.model = this.extractModel(snapshot);
+    this.registrationConfig = this.extractConfig(snapshot);
+    this.options = this.registrationFormConfiguration.getRegistrationFormConfigurationOptions(
```

Comment:
Kind of lengthy, wordy and redundant. Could you rename the method into just `getOptions()`?